scripts/remote-dev-menu.sh

```
@@ -105,6 +114,212 @@ run_diagnostics() {
+  if [[ ! "$choice" =~ ^[0-9]+$ ]]; then
+    return 1
+  fi
+
+  normalized_choice="${choice#"${choice%%[!0]*}"}"
+  if [[ -z "$normalized_choice" ]]; then
+    normalized_choice=0
+  fi
+  if [[ "$normalized_choice" == 0 ]] || (( ${#normalized_choice} > ${#max_choice} )); then
+    return 1
+  fi
+  if (( ${#normalized_choice} == ${#max_choice} )) && [[ "$normalized_choice" > "$max_choice" ]]; then
```

Shellcheck (0.11.0)

[error] > is for string comparisons. Use -gt instead.

(SC2071)